src/dnx/dnx.cpp

```diff
@@ -386,22 +307,34 @@ extern "C" int __stdcall DnxMain(int argc, wchar_t* argv[])
     for (int i = 1; i < argc; ++i)
     {
         //anything without - or -- is appbase or non-dnx command
-        if (argv[i][0] != _T('-'))
+        if (argv[i][0] != _X('-'))
         {
             break;
         }
-        if (StringsEqual(argv[i], _T("--appbase")))
+        if (StringsEqual(argv[i], _X("--appbase")))
         {
             //skip path argument
             ++i;
             continue;
         }
-        if (StringsEqual(argv[i], _T("--debug")))
+        if (StringsEqual(argv[i], _X("--debug")))
         {
             WaitForDebuggerToAttach();
             break;
         }
     }
 
-    return CallApplicationProcessMain(argc, argv);
+    int nExpandedArgc = -1;
+    LPTSTR* ppszExpandedArgv = nullptr;
+    auto expanded = ExpandCommandLineArguments(argc - 1, &(argv[1]), nExpandedArgc, ppszExpandedArgv);
+
+    auto traceWriter = TraceWriter{ IsTracingEnabled() };
+    if (!expanded)
+    {
+        return CallApplicationProcessMain(argc - 1, &argv[1], traceWriter);
+    }
+
+    auto exitCode = CallApplicationProcessMain(nExpandedArgc, ppszExpandedArgv, traceWriter);
```

Comment:
There is no exception here - `CallApplicationProcessMain` does not throw but it catches any exception and turns it to `1` as the `exitCode`.